calico_containers/calico_ctl/status.py

```diff
@@ -35,29 +36,34 @@ def status(arguments):
     this file's docstring with docopt
     :return: None
     """
-    calico_node_info = filter(lambda container: "/calico-node" in
-                              container["Names"],
-                              docker_client.containers())
-    if len(calico_node_info) == 0:
-        print "calico-node container not running"
-    else:
-        print "calico-node container is running. Status: %s" % \
-              calico_node_info[0]["Status"]
-
-        libraries_cmd = docker_client.exec_create("calico-node",
-                                                  ["sh", "-c",
-                                                   "cat libraries.txt"])
-        libraries_out = docker_client.exec_start(libraries_cmd)
-        result = re.search(r"^calico\s*\((.*)\)\s*$", libraries_out,
-                           re.MULTILINE)
-
-        if result is not None:
-            print "Running felix version %s" % result.group(1)
-
-        print "\nIPv4 BGP status"
-        pprint_bird_protocols(4)
-        print "IPv6 BGP status"
-        pprint_bird_protocols(6)
+    try:
+        calico_node_info = filter(lambda container: "/calico-node" in
+                                  container["Names"],
+                                  docker_client.containers())
+        if len(calico_node_info) == 0:
+            print "calico-node container not running"
+        else:
+            print "calico-node container is running. Status: %s" % \
+                  calico_node_info[0]["Status"]
+
+            libraries_cmd = docker_client.exec_create("calico-node",
+                                                      ["sh", "-c",
+                                                       "cat libraries.txt"])
+            libraries_out = docker_client.exec_start(libraries_cmd)
+            result = re.search(r"^calico\s*\((.*)\)\s*$", libraries_out,
+                               re.MULTILINE)
+
+            if result is not None:
+                print "Running felix version %s" % result.group(1)
+
+            print "\nIPv4 BGP status"
+            pprint_bird_protocols(4)
+            print "IPv6 BGP status"
+            pprint_bird_protocols(6)
+    except ConnectionError:
+        print "Docker is not running"
```

Comment:
Can we make this message meaningful to a user that doesn't want to use Docker?  Ditto with other messages.  The messages are factually true, but with multiple runtimes it isn't clear what the user should expect to see.

I'm wondering if we should write some data to an OS file to make `calicoctl` stateful and more helpful.  It would help to know answers to questions like

 - has an admin run `calicoctl node` on this host before?
 - what runtime did they use the last time they ran it?

We could even log all the `calicoctl` commands, which would be super helpful for debugging issues in the field.